### PR TITLE
[Network] Service endpoint policies

### LINKED ISSUE #6932: CLI support for serviceEndpointPolicies
**Is your feature request related to a problem? Please describe.**
This feature is to add policies to service endpoints.

Swagger and PS in review:

https://github.com/Azure/azure-rest-api-specs/pull/3558/files

### PR DESCRIPTION
**Pull request by tjprescott**
**Note:** This is a merge into a dedicated branch for Ignite networking work. It won't pass CI right now and I'm only looking for code review so that specific features can be reviewed in small pieces instead of a huge PR later.

This adds support for service endpoint policies and policy definitions.

Closes #6932. cc/ @chandrasekarsrinivasan 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: azure-cli2017.pyproj [hunk 1]
```diff
@@ -13,8 +13,7 @@
     <LaunchProvider>Standard Python launcher</LaunchProvider>
     <InterpreterId>MSBuild|{54f4b6dc-0859-46dc-99bb-b275c9d0aca3}|$(MSBuildProjectFullPath)</InterpreterId>
     <EnableNativeCodeDebugging>False</EnableNativeCodeDebugging>
-    <CommandLineArguments>
-    </CommandLineArguments>
+    <CommandLineArguments>network vnet subnet create -g tjp-vm --vnet-name vnet1 -n subnet1 --service-endpoints Microsoft.Storage Microsoft.Sql --address-prefix 10.0.0.0/24 --service-endpoint-policy policy1</CommandLineArguments>
```
Comment (yugangw-msft) [inline azure-cli2017.pyproj]: you can back out this change right now, or wait till merge into the dev branch

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/latest/test_network_commands.py [hunk 2]
```diff
@@ -2029,5 +2029,42 @@ def test_network_watcher(self, resource_group, storage_account):
...
+            'vnet': 'vnet1',
+            'subnet': 'subnet1',
+            'rg': 'tjp-vm'
+        })
+
+        self.cmd('network service-endpoint list -l westus')
+
+        # test policy CRUD
+        self.cmd('network service-endpoint policy create -g {rg} -n {policy} --tags test=best')
+        self.cmd('network service-endpoint policy update -g {rg} -n {policy} --tags test=nest')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/latest/test_network_commands.py]: Please add some validations here for this test case.
Also, are you able to get the wheel file from associated Python SDK PR?  If yes, you can put under the `privates` folder, then CI can run normally.

Comment (tjprescott) [reply src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/latest/test_network_commands.py]: Yes, I will add validations. I'm not concerned right now about the CI.

**Comment by tjprescott** [comment]
Closes #7104.

**Pull request merged**